Tree/Binary_search_tree.py

```diff
@@ -0,0 +1,71 @@
+from typing import Any
+from typing import List
+
+
+class BST:
+    def __init__(self, data: Any) -> None:
+        """
+        Time - O(1)
+        Space - O(1)
+        """
+        self.data = data
+        self.left = None
+        self.right = None
+
+    def insert(self, data: Any) -> None:
+        """
+        Time - O(N)
+        Space - O(N)
+        """
+        if data < self.data:
+            if self.left == None:
+                self.left = BST(data)
+            else:
+                self.left.insert(data)
+        else:
+            if self.right == None:
+                self.right = BST(data)
+            else:
+                self.right.insert(data)
+
```

Comment:
Lets make insert work always, No need to initialise root node separately and then start calling insert on it.
```suggestion
    def insert(self, data: Any) -> None:
        """
        Time - O(N)
        Space - O(N)
        """
        if root is None:
          self.root = BinaryTree(data)
         
        if data < self.data:
            if self.left == None:
                self.left = BST(data)
            else:
                self.left.insert(data)
        else:
            if self.right == None:
                self.right = BST(data)
            else:
                self.right.insert(data)
```